python/cudf/cudf/core/dataframe.py

```diff
@@ -2130,30 +2130,10 @@ def _n_largest_or_smallest(self, method, n, columns, keep):
         return df.set_index(self.index.take(new_positions))
 
     def transpose(self):
-        """Transpose index and columns.
-
-        Returns
-        -------
-        a new (ncol x nrow) dataframe. self is (nrow x ncol)
-
-        Notes
-        -----
-        Difference from pandas:
-        Not supporting *copy* because default and only behaviour is copy=True
-        """
-        # Never transpose a MultiIndex - remove the existing columns and
-        # replace with a RangeIndex. Afterward, reassign.
-        columns = self.index.copy(deep=False)
-        index = self.columns.copy(deep=False)
-        if self._num_columns == 0 or self._num_rows == 0:
-            return DataFrame(index=index, columns=columns)
-        # Cython renames the columns to the range [0...ncols]
-        result = self.__class__._from_table(libcudf.transpose.transpose(self))
-        # Set the old column names as the new index
-        result._index = as_index(index)
-        # Set the old index as the new column names
-        result.columns = columns
-        return result
+        raise NotImplementedError(
+            "transpose() is not supported on cudf.DataFrame. "
+            "Consider using DataFrame.to_pandas().transpose()"
```

Comment:
We shouldn't advise using Pandas for transpose. We should point to CuPy instead or transparently handle it with CuPy.

Reply:
This was my suggestion. My bad.